### ISSUE TITLE
tests for findAndLabel and trainMatch

### ISSUE THREAD

**Issue by sonalgoyal** [zinggAI/zingg#901]
we need integration tests for these

**Comment by sania-16** [comment]
These tests have to be in TestExecutorsGeneric.java or in a different file?

**Comment by sonalgoyal** [comment]
You can create a new test extending `TestExecutorGeneric`. Call it `TestExecutorsCompoundPhases` or something and overide `testExecutors`? See if that works.

**Comment by sania-16** [comment]
- we make new testers for them in executor - findandlabeltester and trainMatchtester
- we define them with using object of trainingdatatester and labellertester for validateresults function - override it
- for trainMatch not sure how to do it because of accuracy test - how to proceed?

**Comment by sonalgoyal** [comment]
Let us break up `testexecutorsgeneric`. 
Move 
```
protected abstract TrainingDataFinder<S, D, R, C, T> getTrainingDataFinder() throws ZinggClientException;
    
    protected abstract Labeller<S, D, R, C, T> getLabeller() throws ZinggClientException;
    
    protected abstract Trainer<S, D, R, C, T> getTrainer() throws ZinggClientException;

    protected abstract TrainerValidator<S, D, R, C, T> getTrainerValidator(Trainer<S, D, R, C, T> trainer);

    protected abstract Matcher<S, D, R, C, T> getMatcher() throws ZinggClientException; 

    protected abstract Linker<S, D, R, C, T> getLinker() throws ZinggClientException; 

```
into `TestSingleExecutors` and make it extend `TestExecutorsGeneric`. `TestCompoundExecutors` will extend `TestSingleExecutors`

**Comment by sania-16** [comment]
`TestSingleExecutors` will extend `TestExecutorsGeneric`
`TestCompoundExecutors` will extend `TestSingleExecutors`
`TestExecutorsGeneric` will not extend anything?

`TestSparkExecutors` will extend `TestSingleExecutors`
`TestSparkExecutorsCompound` will extend `TestCompoundExecutors`

and how do we implement methods in `TestSingleExecutors` in `TestExecutorsGeneric`?
by creating an instance as they are abstract classes?

**Comment by sonalgoyal** [comment]
TestExecutorsGeneric - make getExecutors abstract. Move current implementation to TestExecutorsSingle.
